docs/using/troubleshooting.md

```diff
@@ -0,0 +1,310 @@
+
+   For example, to change the ownership of the volume mount to the jovyan user (non-privileged default user in the Docker images):
+
+   ```bash
+   # running in detached mode - can also be run in interactive mode
+   docker run -d \
+       -v <my-vol>:<container-dir> \
+       -p 8888:8888 \
+       --user root \
+       -e CHOWN_EXTRA="<container-dir>" \
+       -e CHOWN_EXTRA_OPTS="-R" \
+       jupyter/minimal-notebook
+   ```
+
+   where:
+
+   - `CHOWN_EXTRA=<some-dir>,<some-other-dir>`: will change the ownership and group of the specified container directory (non-recursive by default). You need to provide full paths starting with `/`.
+   - `CHOWN_EXTRA_OPTS="-R"`: will recursively change the ownership and group of the directory specified in `CHOWN_EXTRA`.
+   - `--user root`: you **must** run the container with the root user to change ownership at runtime.
+
+   now accessing the mount should work as expected:
+
+   ```bash
+   # assuming we mounted the volume in /home/jovyan/stagingarea
+   ls -ld ~/stagingarea
+   # drwxr-xr-x 2 jovyan users 4096 Feb  1 12:55 stagingarea/
+
+   touch stagingarea/kale.txt
+   # jovyan is now the owner of /home/jovyan/stagingarea
+   # ls -la ~/stagingarea/
+   # -rw-r--r-- 1 jovyan users    0 Feb  1 14:41 kale.txt
+   ```
+
+   ```{admonition} Additional notes
+      - If you are mounting your volume inside the `/home/` directory, you can use the `-e CHOWN_HOME=yes` and `CHOWN_HOME_OPTS="-R"` flags
+      instead of the `-e CHOWN_EXTRA` and `-e CHOWN_EXTRA_OPTS` in the example above.
+      - This solution should work in most cases where you have created a docker volume
+      (i.e. using the [`docker volume create --name <my-volume>`command](https://docs.docker.com/storage/volumes/#create-and-manage-volumes)) and mounted it using the`-v` flag in `docker run`.
+   ```
+
+2. **Matching the container's UID/GID with the host's**
+
+   Docker handles mounting host directories differently from mounting volumes, even though the syntax is essentially the same (i.e. `-v`).
+
+   When you initialize a Docker container using the flag `-v`, the host directories are bind-mounted directly into the container.
+   Therefore, the permissions and ownership are copied over and will be **the same** as the ones in your local host
+   (including user ids) which may result in permissions errors when trying to access directories or create/modify files inside.
+
+   Suppose your local user has a `UID` and `GID` of `1234` and `5678`, respectively. To fix the UID discrepancies between your local directories and the container's
+   directories, you can run the container with an explicit `NB_UID` and `NB_GID` to match the that of the local user:
+
+   ```bash
+   docker run -it --rm \
+       --user root \
+       -p 8888:8888 \
+       -e NB_UID=1234 \
+       -e NB_GID=5678 \
+       -v "$(PWD)"/test:/home/jovyan/work \
```

Comment:
This way of getting PWD won't work.
It tries to **execute** `PWD` command, which doesn't exist.
The correct way of doing this is to either use pwd command or the environment variable.

We've always used the env variable like this:

```suggestion
       -v "${PWD}"/test:/home/jovyan/work \
```

Reply:
Other places have this problem as well

Reply:
you are correct there - fixed